src/csharp/SDK/Capture.cs

```diff
@@ -15,124 +26,116 @@ internal Capture(NativeMethods.k4a_capture_t handle)
             this.handle = handle;
         }
 
-        public static Capture Create()
+        private NativeMethods.k4a_capture_t handle;
+
+        private bool disposedValue = false; // To detect redundant calls
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Capture"/> class.
+        /// </summary>
+        public Capture()
         {
+#pragma warning disable CA2000 // Dispose objects before losing scope
             AzureKinectException.ThrowIfNotSuccess(
-                NativeMethods.k4a_capture_create(out NativeMethods.k4a_capture_t handle));
+                NativeMethods.k4a_capture_create(out this.handle));
+#pragma warning restore CA2000 // Dispose objects before losing scope
 
-            return new Capture(handle);
+            // Hook the native allocator and register this object.
+            // .Dispose() will be called on this object when the allocator is shut down.
+            Allocator.Singleton.RegisterForDisposal(this);
         }
 
-
-        
-
-        // This function retrieves a native image handle from the native API.
-        //
-        // If this is the first time calling the property, we construct a new wrapper
-        // If the handle is for an Image we have already constructed a wrapper for, we return the existing wrapper
-        // If the handle is for a different Image, or if the wrapper has already been disposed, we construct a new wrapper and dispose the old one
-        private Image GetImageWrapperAndDisposePrevious(Func<NativeMethods.k4a_capture_t, NativeMethods.k4a_image_t> nativeMethod, 
-            ref Image cache)
+        /// <summary>
+        /// Gets or sets the color image of this capture.
+        /// </summary>
+        /// <remarks>
+        /// Images assigned to this capture are owned by the Capture. When the Capture is disposed, the
+        /// Color Image will be disposed.
+        ///
+        /// By setting the Color image on a Capture, the Capture takes ownership and the Capture will
+        /// dispose the Image when the Capture is disposed.
+        ///
+        /// To get an instance of an Image that lives longer than the capture, call Image.Reference().
```

Comment:
I would add some of the xml directives like <see cref""/> to make sure the documentation knows about the links.

Reply:
I'm going to wait on that until we get the doxygen output going. I'd like to get a preview of what the links end up doing.